docs/standard/serialization/system-text-json-supported-collection-types.md

```diff
@@ -0,0 +1,293 @@
+|----------------------------------------------------------------|---------------|-----------------|
+| <xref:System.Collections.ObjectModel.Collection%601>           | ✔️            | ✔️             |
+| [KeyedCollection\<string, TValue>](xref:System.Collections.ObjectModel.KeyedCollection%602) |✔️|❌|
+| <xref:System.Collections.ObjectModel.ObservableCollection%601> | ✔️            | ✔️             |
+| <xref:System.Collections.ObjectModel.ReadOnlyCollection%601>   | ✔️            | ❌             |
+| [ReadOnlyDictionary\<string, TValue>](xref:System.Collections.ObjectModel.ReadOnlyDictionary%602) |✔️|❌|
+| <xref:System.Collections.ObjectModel.ReadOnlyObservableCollection%601>| ✔️     | ❌             |
+
+::: zone-end
+
+## Custom collections
+
+Any collection type that isn't in one of the preceding namespaces is considered a custom collection. Such types include user-defined types and types defined by ASP.NET Core. For example, <xref:Microsoft.Extensions.Primitives?displayProperty=fullName> is in this group.
+
+All custom collections (everything that derives from `IEnumerable`) are supported for serialization, as long as their element types are supported.
+
+### Custom collections with deserialization support
+
+A custom collection is supported for deserialization if it:
+
+::: zone pivot="dotnet-5-0"
+
+* Isn't an interface or abstract.
+* Has a parameterless constructor.
+* Implements or inherits one or more of the following interfaces or classes:
+  * <xref:System.Collections.Concurrent.ConcurrentQueue%601>
+  * <xref:System.Collections.Concurrent.ConcurrentStack%601> \*
+  * <xref:System.Collections.Generic.ICollection%601>
+  * <xref:System.Collections.IDictionary>
+  * <xref:System.Collections.Generic.IDictionary%602>
+  * <xref:System.Collections.IList>
+  * <xref:System.Collections.Generic.IList%601>
+  * <xref:System.Collections.Queue>
+  * <xref:System.Collections.Generic.Queue%601>
+  * <xref:System.Collections.Stack> \*
+  * <xref:System.Collections.Generic.Stack%601> \*
+* Contains element types that are supported by <xref:System.Text.Json.JsonSerializer>.
+
+::: zone-end
+
+::: zone pivot="dotnet-core-3-1"
+
+* Isn't an interface or abstract.
+* Has a parameterless constructor.
+* Implements or inherits one or more of the following interfaces or classes:
+  * <xref:System.Collections.Concurrent.ConcurrentQueue%601>
+  * <xref:System.Collections.Concurrent.ConcurrentStack%601> \*
+  * <xref:System.Collections.Generic.ICollection%601>
+  * <xref:System.Collections.IDictionary>
+  * [IDictionary\<string, TValue>](xref:System.Collections.Generic.IDictionary%602)
+  * <xref:System.Collections.IList>
+  * <xref:System.Collections.Generic.IList%601>
+  * <xref:System.Collections.Queue>
+  * <xref:System.Collections.Generic.Queue%601>
+  * <xref:System.Collections.Stack> \*
+  * <xref:System.Collections.Generic.Stack%601> \*
+* Contains element types that are supported by <xref:System.Text.Json.JsonSerializer>.
```

Comment:
Consider putting this bullet before "Implements or inherits..." for readability and so the asterisk matches up better.

Reply:
Good idea, thanks!